js/components/graph/Edge.js

```diff
@@ -8,46 +8,33 @@ import { refLookUp } from "../common/utils"
 export default class Edge extends React.Component {
   constructor(props) {
     super(props)
-    this.state = { status: "inactive" }
   }
 
-  /**
-   * Update the status of the Edge, based on the status of the Node/Bool it points from/to
-   */
-  updateStatus = () => {
-    var source = refLookUp(this.props.source, this.props.svg)
-    var target = refLookUp(this.props.target, this.props.svg)
-    if (source === undefined || target === undefined) {
+  updateStatus(status) {
+    const sourceNode = refLookUp(this.props.source, this.props.svg)
+    const targetNode = refLookUp(this.props.target, this.props.svg)
+    if (sourceNode === undefined || targetNode === undefined) {
       return
     }
-
-    if (
-      !source.isSelected() &&
-      ((target.props.className === "bool" && target.props.status === "missing") ||
-        (target.props.className !== "bool" && target.state.status === "missing"))
-    ) {
-      this.setState({ status: "missing" })
-    } else if (!source.isSelected()) {
-      this.setState({ status: "inactive" })
-    } else if (!target.isSelected()) {
-      this.setState({ status: "takeable" })
-    } else {
-      this.setState({ status: "active" })
-    }
-  }
-  /**
-   *
-    After each render beyond the initial, check if the edge's state has changed. If so,
-    notify the state of EdgeGroup with updateEdgeStatus.
-   * @param {*} prevProps
-   * @param {Object} prevState The state of this object from the previous render
-   */
-  componentDidUpdate(prevProps, prevState) {
-    if (this.state.status !== prevState.status) {
-      this.props.updateEdgeStatus(this.props.edgeID, this.state.status)
+    if (!status) {
+      if (
+        !sourceNode.isSelected() &&
+        ((targetNode.props.className === "bool" &&
+          targetNode.props.status === "missing") ||
+          (targetNode.props.className !== "bool" &&
+            targetNode.state.status === "missing"))
+      ) {
+        status = "missing"
+      } else if (!sourceNode.isSelected()) {
+        status = "inactive"
+      } else if (!targetNode.isSelected()) {
+        status = "takeable"
+      } else {
+        status = "active"
+      }
     }
+    this.props.updateEdgeStatus(status, this.props.edgeID)
   }
-
```

Comment:
Revert this change

Reply:
Should I move the logic from line 18 to 37 into Graph like we discussed?

Reply:
@leowrites you don't need to do that for this PR. It's better to get this one merged in first, and then do subsequent updates in the Node PR.